project/driver_controls/src/driver_io.c

```diff
@@ -0,0 +1,106 @@
+#include <stddef.h>
+
+#include "gpio_it.h"
+#include "driver_io.h"
+#include "driver_callback.h"
+
+// Arbitrary thresholds for gas pedal
+#define COAST_THRESHOLD 1000
+#define DRIVE_THRESHOLD 3000
+
+typedef struct DriverIOSettings {
+  GPIOAddress address;
+  GPIODir direction;
+  GPIOAltFn alt_function;
+  InterruptEdge edge;
+  DriverIOCallback callback;
+} DriverIOSettings;
+
+// Lookup table to map specific input devices to specific pins
+static DriverIOData s_devices[NUM_DRIVER_IO_PIN] = {
```

Comment:
What's the point of all the `DRIVER_IO_PIN_x`s? Is that information used anywhere? Why not just use `DRIVER_IO_x`?

I'm not really convinced that this driver IO implementation is really reducing complexity at all. It seems to be a lot more work to manage.